### ISSUE TITLE
Spurious failures checking out repo

### ISSUE THREAD

**Issue by pitrou** [actions/checkout#188]
Sometimes the git checkout step fails doing the git checkout, even though the referenced commit does exist:
https://github.com/apache/arrow/pull/6632/checks?check_run_id=511663097
```
git checkout --progress --force 1a41f432e66a2a5c47f78aa7586deec049416fac
##[error]fatal: reference is not a tree: 1a41f432e66a2a5c47f78aa7586deec049416fac
```

I've witnessed this several times.

**Comment by ericsciple** [comment]
@pitrou this sounds like a race condition that was fixed with `actions/checkout@v2`. Are you using `actions/checkout@v1`?

**Comment by pitrou** [comment]
We did. But we just fixed it in #6634. Thank you!

**Comment by ericsciple** [comment]
Thanks! Sounds like https://github.com/actions/checkout/issues/23
